integration/local-gadget/integration_test.go

```diff
@@ -0,0 +1,256 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+
+	. "github.com/kinvolk/inspektor-gadget/integration"
+	containercollection "github.com/kinvolk/inspektor-gadget/pkg/container-collection"
+	bindTypes "github.com/kinvolk/inspektor-gadget/pkg/gadgets/trace/bind/types"
+)
+
+const (
+	ContainerRuntimeDocker     = "docker"
+	ContainerRuntimeContainerd = "containerd"
+	ContainerRuntimeCRIO       = "cri-o"
+)
+
+var supportedContainerRuntimes = []string{ContainerRuntimeDocker, ContainerRuntimeContainerd, ContainerRuntimeCRIO}
+
+var (
+	integration      = flag.Bool("integration", false, "run integration tests")
+	containerRuntime = flag.String("container-runtime", "", "allows to do validation for expected runtime in the tests")
+)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	if !*integration {
+		fmt.Println("Skipping integration test.")
+		os.Exit(0)
+	}
+
+	if *containerRuntime != "" {
+		found := false
+		for _, val := range supportedContainerRuntimes {
+			if *containerRuntime == val {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			fmt.Fprintf(os.Stderr, "Error: invalid argument '-container-runtime': %q. Valid values: %s\n",
+				*containerRuntime, strings.Join(supportedContainerRuntimes, ", "))
+			os.Exit(-1)
+		}
+	} else {
+		fmt.Println("Error: '-container-runtime' must be specified")
+		os.Exit(1)
+	}
```

Comment:
nit: You can move this to L36 and avoid the `*containerRuntime != ""` check. Something like:

```go 
	if *containerRuntime == "" {
		fmt.Println("Error: '-container-runtime' must be specified")
		os.Exit(1)
	}

	found := false
	for _, val := range supportedContainerRuntimes {
		if *containerRuntime == val {
			found = true
			break
		}
	}

	if !found {
		fmt.Fprintf(os.Stderr, "Error: invalid argument '-container-runtime': %q. Valid values: %s\n",
			*containerRuntime, strings.Join(supportedContainerRuntimes, ", "))
		os.Exit(-1)
	}
```